src/google/protobuf/compiler/parser.cc

```diff
@@ -615,6 +615,7 @@ bool Parser::Parse(io::Tokenizer* input, FileDescriptorProto* file) {
   input_ = input;
   had_errors_ = false;
   syntax_identifier_.clear();
+  input->set_require_valid_utf8(true);
```

Comment:
Is this unconditional change intended?

Reply:
This is what actually enables enforcement that proto sources have valid UTF8. But I am happy to make it conditional. Do you think this should be behind a compile-time #define flag? Or maybe it could be disabled via protoc command-line flag? I think it's appropriate that the default behavior, at least for open-source builds, be that this does enforce UTF8. But I'm not sure what the best "idiomatic" way to make it conditional would be. Advice?

Reply:
I added a comment. I'm interested in advice about the most appropriate way to make this conditional.